NamedPipeWrapper/NamedPipeServer.cs

```diff
@@ -236,13 +276,39 @@ public void Stop()
 				}
 			}
 
+			if (_server != null && _server.IsConnected)
+			{
+				_server.Close();
+			}
+
 			// If background thread is still listening for a client to connect,
 			// initiate a dummy connection that will allow the thread to exit.
 			var dummyClient = new NamedPipeClient<TRead, TWrite>(_pipeName);
 			dummyClient.Start();
 			dummyClient.WaitForConnection(TimeSpan.FromSeconds(2));
 			dummyClient.Stop();
 			dummyClient.WaitForDisconnection(TimeSpan.FromSeconds(2));
+
+		}
+
+		public void Read()
```

Comment:
Why do we need a `Read()` method? Isn't that being handled in the `ClientOnReceiveMessage()` event handler?